modules/tensor_mechanics/src/materials/ComputeIncrementalBeamStrain.C

```diff
@@ -0,0 +1,590 @@
+  _mech_rot_strain_increment[_qp](1) = _grad_disp_0_local_t(0) * _Az[_qp] -
+                                       _grad_rot_0_local_t(2) * J +
+                                       _grad_rot_0_local_t(1) * _Iz[_qp];
+  _mech_rot_strain_increment[_qp](2) = _grad_disp_0_local_t(0) * _Ay[_qp] -
+                                       _grad_rot_0_local_t(2) * _Iy[_qp] +
+                                       _grad_rot_0_local_t(1) * J;
+
+  if (_large_strain)
+  {
+    _mech_disp_strain_increment[_qp](0) +=
+        0.5 *
+        ((Utility::pow<2>(_grad_disp_0_local_t(0)) + Utility::pow<2>(_grad_disp_0_local_t(1)) +
+          Utility::pow<2>(_grad_disp_0_local_t(2))) *
+             _area[_qp] +
+         Utility::pow<2>(_grad_rot_0_local_t(2)) * _Iy[_qp] +
+         Utility::pow<2>(_grad_rot_0_local_t(1)) * _Iz[_qp] +
+         Utility::pow<2>(_grad_rot_0_local_t(0)) * (_Iy[_qp] + _Iz[_qp]));
+    _mech_disp_strain_increment[_qp](1) += (-_avg_rot_local_t(2) * _grad_disp_0_local_t(0) +
+                                            _avg_rot_local_t(0) * _grad_disp_0_local_t(2)) *
+                                           _area[_qp];
+    _mech_disp_strain_increment[_qp](2) += (_avg_rot_local_t(1) * _grad_disp_0_local_t(0) -
+                                            _avg_rot_local_t(0) * _grad_disp_0_local_t(1)) *
+                                           _area[_qp];
+
+    _mech_rot_strain_increment[_qp](0) += -_avg_rot_local_t(1) * _grad_rot_0_local_t(2) * _Iy[_qp] +
+                                          _avg_rot_local_t(2) * _grad_rot_0_local_t(1) * _Iz[_qp];
+    _mech_rot_strain_increment[_qp](1) += (_grad_disp_0_local_t(0) * _grad_rot_0_local_t(1) -
+                                           _grad_disp_0_local_t(1) * _grad_rot_0_local_t(0)) *
+                                          _Iz[_qp];
+    _mech_rot_strain_increment[_qp](2) += (_grad_disp_0_local_t(2) * _grad_rot_0_local_t(0) -
+                                           _grad_disp_0_local_t(0) * _grad_rot_0_local_t(2)) *
+                                          _Iy[_qp];
+  }
+
+  _total_disp_strain[_qp] = _total_rotation[0].transpose() * _mech_disp_strain_increment[_qp] +
+                            _total_disp_strain_old[_qp];
+  _total_rot_strain[_qp] = _total_rotation[0].transpose() * _mech_rot_strain_increment[_qp] +
+                           _total_disp_strain_old[_qp];
+
+  // Convert eigenstrain increment from global to beam local coordinate system and remove eigen
+  // strain increment
+  for (unsigned int i = 0; i < _eigenstrain_names.size(); ++i)
+  {
+    _mech_disp_strain_increment[_qp] -=
+        _total_rotation[0] * ((*_disp_eigenstrain[i])[_qp] - (*_disp_eigenstrain_old[i])[_qp]) *
+        _area[_qp];
+    _mech_rot_strain_increment[_qp] -=
+        _total_rotation[0] * ((*_rot_eigenstrain[i])[_qp] - (*_rot_eigenstrain_old[i])[_qp]);
+  }
+}
+
+void
+ComputeIncrementalBeamStrain::computeJacobian()
+{
```

Comment:
This is analogous to the material stiffness calculated by the stress calculator. Could this be moved to `computeBeamResultants`? Also, I'd call it something other than `computeJacobian` to avoid confusion with the Kernel's `computeJacobian`.

Reply:
For the small strain scenario, the stiffness matrix depends only on the material properties and the geometry of the beam. But for the large strain scenario, the displacements of the beam are also needed to get the stiffness matrix. So if I move the stiffness matrix calculation to ComputeBeamResultants, I will have either redo some of the displacement calculation or transfer those over from the ComputeIncrementalBeamStrain to ComputeBeamResultants.

Reply:
I think this is pretty analogous to what happens in `StressDivergenceTensors::computeFiniteDeformJacobian()`.  I had suggested that the stiffness be computed in `ComputeBeamResultants`, but I guess it would really need to be moved to the Kernel to be consistent with that.  Aside from this, I think this code is ready to go, and I'm going to merge it.

I would however still like to look into whether we can make this more consistent with how the solid mechanics equivalent works. To avoid re-computing displacements or other quantities, maybe we can just make them material properties that the Kernel accesses.

Reply:
Thanks, Ben. I will edit and submit the other PRs and when you and Al are reviewing the other PRs, I will look into whether this can be made more consistent with the solid mechanics equivalent.